src/Avalonia.Base/Input/Focus/FocusInputDeviceKind.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace Avalonia.Input
+{
+    /// <summary>
+    /// Specifies input device types from which input events are received
+    /// </summary>
+    public enum FocusInputDeviceKind
```

Comment:
you could add software or automation

Reply:
For "software" "Programmatic" term is used usually.

Reply:
UWP uses "None" for that:
```
No input. Used only when the focus is moved programmatically.
```
https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.input.focusinputdevicekind?view=winrt-22621